### PR TITLE
Update pace to use python 3.11.9

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
Updates to constraints and requirements to enable the use of python 3.11.9 

**How Has This Been Tested?**
Tested using the main unit tests in `tests/main`

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

File: .github/workflows/main_unit_tests.yml
Comment (mlee03) [inline .github/workflows/main_unit_tests.yml:1]: This file is no longer needed, the tests triggered with this workflow now run in containers

Comment (fmalatino) [reply .github/workflows/main_unit_tests.yml:1]: Sounds good will remove

**Comment by mlee03** [comment]
Could we use this PR to test the updated images?
Two additional changes are required, the tag has to be updated to 3.11.9 in [here](https://github.com/NOAA-GFDL/pace/blob/develop/.github/workflows/main_unit_tests_mpich.yaml#L15) and [here](https://github.com/NOAA-GFDL/pace/blob/develop/.github/workflows/main_unit_tests_openmpi.yaml#L15)

**Comment by fmalatino** [comment]
> Could we use this PR to test the updated images? Two additional changes are required, the tag has to be updated to 3.11.9 in [here](https://github.com/NOAA-GFDL/pace/blob/develop/.github/workflows/main_unit_tests_mpich.yaml#L15) and [here](https://github.com/NOAA-GFDL/pace/blob/develop/.github/workflows/main_unit_tests_openmpi.yaml#L15)

Can do. Will update and re-push

**Review by mlee03 — APPROVED** [review]
Yay CI works.

**Review by xyuan — APPROVED** [review]
thanks, this is great job

**Comment by FlorianDeconinck** [comment]
NB: this will break orchestration, we have failures with py 3.11.X. I am fixing it DaCe side. Should not block for now, we have CI coverage issues anyway we will discuss on Friday

**Review by oelbert — APPROVED** [review]
👍

**Pull request merged**